staging/src/k8s.io/apiserver/pkg/util/shufflesharding/shufflesharding.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package shufflesharding
+
+import (
+	"errors"
+	"math"
+)
+
+const maxHashBits = 60
+
+// ValidateParameters can validate parameters for shuffle sharding
+// in a fast but approximate way, including numQueues and handSize
+// Algorithm: maxHashValue >= numQueues^handSize
+func ValidateParameters(numQueues, handSize uint32) bool {
```

Comment:
let's remove the validation from this pull. it's done in https://github.com/kubernetes/kubernetes/pull/80257/commits/2d62c16076f608383f4980b721925d3b831f6467

Reply:
It's better to have validation and shuffle functions in one go package?

Reply:
With the validation function here public, the registry strategy for PriorityLevelConfigurationSpec could invoke it.